### PR TITLE
[DIR-1822] Integrate opensearch

### PR DESCRIPTION
**Pull request by h8718**
## Description

to enable use:
```yaml
opensearch:
  install: true
nats:
  install: true
```

in the helm-values on install. 

```
Test Suites: 98 passed, 98 total
Tests:       778 passed, 778 total
Snapshots:   0 total
Time:        251.543 s
Ran all test suites matching /\//i.
npm notice 
npm notice New minor version of npm available! 10.5.2 -> 10.9.1
npm notice Changelog: https://github.com/npm/cli/releases/tag/v10.9.1
npm notice Run npm install -g npm@10.9.1 to update!
npm notice 
```

Do this on cluster setup if you use Linkerd:
```
kubectl annotate namespace direktiv \
  config.linkerd.io/opaque-ports="4222,6222,8222,9200,9300,9600"
```

and ensure to run: 
```
sysctl -w vm.max_map_count=262144
```
on all kubernetes nodes

## Checklist

- [x] Documentation updated if required
- [x] Test coverage is appropriate
      
## Checklist Internal

- [x] Linear issue linked (e.g. [DIR-XXXX] pull request title)
- [x] Has the PR been labeled

### CODE REVIEW

File: pkg/flow/server.go [hunk 1]
```diff
@@ -634,7 +654,71 @@ func (srv *server) publishDemoMessage(subject, msg string) error {
 	if err != nil {
```
Comment (jensg-st) [inline pkg/flow/server.go:654]: why are we publishing a demo message? Is that required?

File: pkg/flow/server.go [hunk 2]
```diff
@@ -634,7 +654,71 @@ func (srv *server) publishDemoMessage(subject, msg string) error {
...
+		return nil, fmt.Errorf("OpenSearch connection test failed: %w", err)
+	}
+	defer res.Body.Close()
+
+	slog.Info("Connected to OpenSearch", "info", res.String())
+
+	return client, nil
+}
+
+func (srv *server) demoIndexDocument(ctx context.Context, index string, docID string, doc string) error {
```
Comment (jensg-st) [inline pkg/flow/server.go:703]: Is that required?

File: pkg/flow/server.go (hunk 1 above)
Comment (h8718) [reply pkg/flow/server.go:649]: no this was just to demo that it's working

Comment (h8718) [reply pkg/flow/server.go:649]: thanks for pointing this out i removed it.
it's a demo how to use it just as the name suggest, it's not required, was just a placeholder ntil we have something real

**Review by jensg-st — APPROVED** [review]

**Review by jensg-st — APPROVED** [review]

**Pull request merged**